pkg/metrics/instance/instance_test.go

```diff
@@ -245,6 +247,82 @@ func TestInstance(t *testing.T) {
 	})
 }
 
+func TestDisabledKeepAlives(t *testing.T) {
+	// When we DisableKeepAlives the Connection header should be set to 'close'
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, r.Header.Get("Connection"), "close")
+	})
+	httpSrv := httptest.NewServer(handler)
+	defer httpSrv.Close()
+
+	walDir, err := ioutil.TempDir(os.TempDir(), "wal")
+	require.NoError(t, err)
+	defer os.RemoveAll(walDir)
+
+	globalConfig := getTestGlobalConfig(t)
+	cfg := getTestConfig(t, &globalConfig, httpSrv.Listener.Addr().String())
+	cfg.DisableKeepAlives = true
+
+	mockStorage := mockWalStorage{series: make(map[storage.SeriesRef]int), directory: walDir}
+	newWal := func(_ prometheus.Registerer) (walStorage, error) { return &mockStorage, nil }
+
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	inst, err := newInstance(cfg, nil, logger, newWal)
+	require.NoError(t, err)
+	runInstance(t, inst)
+
+	// Wait until mockWalStorage has had a series added to it.
+	test.Poll(t, 30*time.Second, true, func() interface{} {
+		mockStorage.mut.Lock()
+		defer mockStorage.mut.Unlock()
+		return len(mockStorage.series) > 0
+	})
+}
+
+func TestIdleConnectionTimeouts(t *testing.T) {
```

Comment:
IMO we don't need to test Prometheus code and we don't really need tests for this since we can tell at a glance the options get passed properly.